<?xml version="1.0" encoding="UTF-8"?>
<pd:ProcessDefinition xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:ns="http://www.tibco.com/namespaces/tnt/plugins/jms" xmlns:ns5="http://www.tibco.com/CommonLE2/namespace/Public/Common/RequestHeader.xsd" xmlns:ns12="http://www.tibco.com/CommonLE2/Private/Common" xmlns:pfx="http://www.tibco.com/CommonLE2/namespace/Exception" xmlns:pd="http://xmlns.tibco.com/bw/process/2003" xmlns:ns2="http://www.tibco.com/CommonLE/namespace/Common" xmlns:ns1="http://www.tibco.com/pe/EngineTypes" xmlns:ns4="http://www.tibco.com/pe/DeployedVarsType" xmlns:ns3="http://www.tibco.com/CommonLE2/namespace/private/Common" xmlns:pfx5="http://www.tibco.com/CommonLE2/namespace/Public/Exception/ExceptionRequest.xsd" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:pfx4="com/tibco/pe/commands" xmlns:tib="http://www.tibco.com/bw/xslt/custom-functions" xmlns:pfx3="http://www.tibco.com/pe/GenerateErrorActivity/InputSchema" xmlns:pfx2="http://www.tibco.com/CommonLE2/namespace/Public/Exception" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
    <xsd:import namespace="http://www.tibco.com/CommonLE2/namespace/private/Common" schemaLocation="/SharedResources/SchemaDefinitions/Private/Common/LocalExceptions.xsd"/>
    <xsd:import namespace="http://www.tibco.com/CommonLE2/namespace/Public/Exception/ExceptionRequest.xsd" schemaLocation="/SharedResources/SchemaDefinitions/Public/Exception/ExceptionRequest.xsd"/>
    <pd:name>ExceptionManagement/ExceptionServices/RV Channel.process</pd:name>
    <pd:description>Receiving an exception request using RV transport. 
The exception request data must be in ExceptionRequest Schema. 
Exception requests that do not conform to this scheam are ignored. 
This process does not perform the main processing logic. </pd:description>
    <pd:startName>ExceptionRequest Subscriber</pd:startName>
    <pd:startX>0</pd:startX>
    <pd:startY>0</pd:startY>
    <pd:returnBindings/>
    <pd:starter name="ExceptionRequest Subscriber">
        <pd:type>com.tibco.plugin.tibrv.RVEventSource</pd:type>
        <pd:resourceType>ae.rvpalette.RVEventSource</pd:resourceType>
        <pd:x>57</pd:x>
        <pd:y>43</pd:y>
        <config>
            <wantsValidationForOutput>true</wantsValidationForOutput>
            <wantsFiltrationForOutput>false</wantsFiltrationForOutput>
            <wantsXMLCompliantFieldNames>false</wantsXMLCompliantFieldNames>
            <wantsRawRVObjectForOutput>false</wantsRawRVObjectForOutput>
            <xmlEncoding>false</xmlEncoding>
            <sharedChannel>/SharedResources/Connections/RV/RV Exception.rvtransport</sharedChannel>
            <subject>%%RVSubjectNames/ExceptionRequestSubject%%</subject>
            <xsdString>
                <xsd:element name="ExceptionRequest" type="xsd:string"/>
            </xsdString>
        </config>
        <pd:inputBindings/>
    </pd:starter>
    <pd:endName>End</pd:endName>
    <pd:endX>567</pd:endX>
    <pd:endY>195</pd:endY>
    <pd:errorSchemas>
        <Catch-JMS-Connection-Error ref="ns3:JMSError"/>
    </pd:errorSchemas>
    <pd:processVariables/>
    <pd:targetNamespace>http://xmlns.example.com/1284579944158</pd:targetNamespace>
    <pd:activity name="Catch DB Error">
        <pd:type>com.tibco.pe.core.CatchActivity</pd:type>
        <pd:resourceType>ae.activities.catch</pd:resourceType>
        <pd:x>47</pd:x>
        <pd:y>155</pd:y>
        <pd:handler>true</pd:handler>
        <config>
            <catchAll>false</catchAll>
            <fault>localname=DBError namespace=http://www.tibco.com/CommonLE2/namespace/private/Common</fault>
        </config>
        <pd:inputBindings/>
    </pd:activity>
    <pd:activity name="Confirm">
        <pd:type>com.tibco.pe.core.ConfirmActivity</pd:type>
        <pd:resourceType>activity.confirm</pd:resourceType>
        <pd:description>Using late comfirm based on performance and data duplication trade off. 
As performance is more important, the late confim is used. As a result, it is
possible to have duplicated excption records. </pd:description>
        <pd:x>407</pd:x>
        <pd:y>35</pd:y>
        <config>
            <ConfirmEvent>ExceptionRequest Subscriber</ConfirmEvent>
        </config>
        <pd:inputBindings/>
    </pd:activity>
    <pd:activity name="Confirm-1">
        <pd:type>com.tibco.pe.core.ConfirmActivity</pd:type>
        <pd:resourceType>activity.confirm</pd:resourceType>
        <pd:x>393</pd:x>
        <pd:y>315</pd:y>
        <config>
            <ConfirmEvent>ExceptionRequest Subscriber</ConfirmEvent>
        </config>
        <pd:inputBindings/>
    </pd:activity>
    <pd:activity name="Handle Local Exception">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>251</pd:x>
        <pd:y>203</pd:y>
        <config>
            <processName>/Common/Local Exception Handler.process</processName>
        </config>
        <pd:inputBindings>
            <ns2:Exception>
                <ErrorMessage>
                    <xsl:value-of select="$_error/ns1:ErrorReport/Msg"/>
                </ErrorMessage>
                <ErrorCode>
                    <xsl:value-of select="$_error/ns1:ErrorReport/MsgCode"/>
                </ErrorCode>
                <ns2:Error>
                    <xsl:for-each select="$_error/ns1:ErrorReport">
                        <ns2:ErrorReport>
                            <StackTrace>
                                <xsl:value-of select="StackTrace"/>
                            </StackTrace>
                            <Msg>
                                <xsl:value-of select="Msg"/>
                            </Msg>
                            <FullClass>
                                <xsl:value-of select="FullClass"/>
                            </FullClass>
                            <Class>
                                <xsl:value-of select="Class"/>
                            </Class>
                            <ProcessStack>
                                <xsl:value-of select="ProcessStack"/>
                            </ProcessStack>
                            <MsgCode>
                                <xsl:choose>
                                    <xsl:when test="exists(MsgCode)">
                                        <xsl:value-of select="MsgCode"/>
                                    </xsl:when>
                                    <xsl:otherwise>
                                        <xsl:attribute name="xsi:nil">true</xsl:attribute>
                                    </xsl:otherwise>
                                </xsl:choose>
                            </MsgCode>
                        </ns2:ErrorReport>
                    </xsl:for-each>
                </ns2:Error>
                <ns2:Data>
                    <xsl:value-of select="tib:render-xml($_error/ns1:ErrorReport/Data)"/>
                </ns2:Data>
            </ns2:Exception>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Invoke Exception Management Process">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>237</pd:x>
        <pd:y>43</pd:y>
        <config/>
        <pd:inputBindings>
            <ExceptionRequest>
                <xsl:value-of select="$ExceptionRequest-Subscriber/ActivityOutput/body/ExceptionRequest"/>
            </ExceptionRequest>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Suspend Exception Request Starter">
        <pd:type>com.tibco.pe.core.EngineCommandActivity</pd:type>
        <pd:resourceType>ae.activities.enginecommand</pd:resourceType>
        <pd:description>Receiving a log request using JMS transport. 
The log request data must be in LogRequest Schema. 
Log requests that do not conform to this schema are ignored. 
This process delegates the main processing logic to a subprocess</pd:description>
        <pd:x>389</pd:x>
        <pd:y>203</pd:y>
        <config>
            <command>SuspendProcessStarter</command>
            <processDefinition>/ExceptionManagement/ExceptionServices/RV Channel.process</processDefinition>
        </config>
        <pd:inputBindings>
            <pfx4:input/>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Catch All">
        <pd:type>com.tibco.pe.core.CatchActivity</pd:type>
        <pd:resourceType>ae.activities.catch</pd:resourceType>
        <pd:x>166</pd:x>
        <pd:y>318</pd:y>
        <pd:handler>true</pd:handler>
        <config>
            <fault>localname=ExceptionConfigurationNotFound namespace=http://www.tibco.com/CommonLE2/namespace/private/Common</fault>
            <catchAll>true</catchAll>
        </config>
        <pd:inputBindings/>
    </pd:activity>
    <pd:activity name="Catch JMS Error">
        <pd:type>com.tibco.pe.core.CatchActivity</pd:type>
        <pd:resourceType>ae.activities.catch</pd:resourceType>
        <pd:x>51</pd:x>
        <pd:y>265</pd:y>
        <pd:handler>true</pd:handler>
        <config>
            <catchAll>false</catchAll>
            <fault>localname=JMSError namespace=http://www.tibco.com/CommonLE2/namespace/private/Common</fault>
        </config>
        <pd:inputBindings/>
    </pd:activity>
    <pd:activity name="Local Exception Handling">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>278</pd:x>
        <pd:y>324</pd:y>
        <config>
            <processName>/Common/Local Exception Handler.process</processName>
        </config>
        <pd:inputBindings>
            <ns2:Exception>
                <xsl:if test="$_error/ns1:ErrorReport/Msg">
                    <ErrorMessage>
                        <xsl:value-of select="$_error/ns1:ErrorReport/Msg"/>
                    </ErrorMessage>
                </xsl:if>
                <xsl:if test="$_error/ns1:ErrorReport/MsgCode">
                    <ErrorCode>
                        <xsl:value-of select="$_error/ns1:ErrorReport/MsgCode"/>
                    </ErrorCode>
                </xsl:if>
                <xsl:for-each select="$_error">
                    <ns2:Error>
                        <xsl:for-each select="ns1:ErrorReport">
                            <ns2:ErrorReport>
                                <StackTrace>
                                    <xsl:value-of select="StackTrace"/>
                                </StackTrace>
                                <Msg>
                                    <xsl:value-of select="Msg"/>
                                </Msg>
                                <FullClass>
                                    <xsl:value-of select="FullClass"/>
                                </FullClass>
                                <Class>
                                    <xsl:value-of select="Class"/>
                                </Class>
                                <ProcessStack>
                                    <xsl:value-of select="ProcessStack"/>
                                </ProcessStack>
                                <xsl:if test="MsgCode">
                                    <MsgCode>
                                        <xsl:value-of select="MsgCode"/>
                                    </MsgCode>
                                </xsl:if>
                            </ns2:ErrorReport>
                        </xsl:for-each>
                    </ns2:Error>
                </xsl:for-each>
            </ns2:Exception>
        </pd:inputBindings>
    </pd:activity>
    <pd:transition>
        <pd:from>Invoke Exception Management Process</pd:from>
        <pd:to>Confirm</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Confirm</pd:from>
        <pd:to>End</pd:to>
        <pd:lineType>S-Shaped</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Catch DB Error</pd:from>
        <pd:to>Handle Local Exception</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Handle Local Exception</pd:from>
        <pd:to>Suspend Exception Request Starter</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Suspend Exception Request Starter</pd:from>
        <pd:to>End</pd:to>
        <pd:lineType>Straight</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Confirm-1</pd:from>
        <pd:to>End</pd:to>
        <pd:lineType>S-Shaped</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Catch All</pd:from>
        <pd:to>Local Exception Handling</pd:to>
        <pd:lineType>Straight</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Catch JMS Error</pd:from>
        <pd:to>Handle Local Exception</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>ExceptionRequest Subscriber</pd:from>
        <pd:to>Invoke Exception Management Process</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Local Exception Handling</pd:from>
        <pd:to>Confirm-1</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
</pd:ProcessDefinition>